packages/backend-api/src/Controller/V1/Product/ProductDataFactory.php

```diff
@@ -0,0 +1,94 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Shopsys\BackendApiBundle\Controller\V1\Product;
+
+use Ramsey\Uuid\Uuid;
+use Shopsys\BackendApiBundle\Component\DataSetter\ApiDataSetter;
+use Shopsys\FrameworkBundle\Model\Product\Availability\AvailabilityFacade;
+use Shopsys\FrameworkBundle\Model\Product\Product;
+use Shopsys\FrameworkBundle\Model\Product\ProductData;
+use Shopsys\FrameworkBundle\Model\Product\ProductDataFactoryInterface as BaseProductDataFactoryInterface;
+
+/**
+ * @experimental
+ */
+class ProductDataFactory implements ProductDataFactoryInterface
+{
+    /**
+     * @var \Shopsys\FrameworkBundle\Model\Product\ProductDataFactoryInterface
+     */
+    protected $productDataFactory;
+
+    /**
+     * @var \Shopsys\FrameworkBundle\Model\Product\Availability\AvailabilityFacade
+     */
+    protected $availabilityFacade;
+
+    /**
+     * @var \Shopsys\BackendApiBundle\Component\DataSetter\ApiDataSetter
+     */
+    protected $apiDataSetter;
+
+    /**
+     * @param \Shopsys\FrameworkBundle\Model\Product\ProductDataFactoryInterface $productDataFactory
+     * @param \Shopsys\FrameworkBundle\Model\Product\Availability\AvailabilityFacade $availabilityFacade
+     * @param \Shopsys\BackendApiBundle\Component\DataSetter\ApiDataSetter $apiDataSetter
+     */
+    public function __construct(
+        BaseProductDataFactoryInterface $productDataFactory,
+        AvailabilityFacade $availabilityFacade,
+        ApiDataSetter $apiDataSetter
+    ) {
+        $this->productDataFactory = $productDataFactory;
+        $this->availabilityFacade = $availabilityFacade;
+        $this->apiDataSetter = $apiDataSetter;
+    }
+
+    /**
+     * @param array $productApiData
+     * @param string|null $uuid
+     * @return \Shopsys\FrameworkBundle\Model\Product\ProductData
+     */
+    public function createFromApi(array $productApiData, ?string $uuid = null): ProductData
+    {
+        $productData = $this->productDataFactory->create();
+        $productData->uuid = $uuid ?: Uuid::uuid4()->toString();
+        $productData->availability = $this->availabilityFacade->getDefaultInStockAvailability();
```

Comment:
Does this mean any created product will be by default "In stock" without any way how to change it? 

I don't like it. I would like to have a way to set availability (even if it means I need to find proper value elsewhere)

Reply:
@PetrHeinz told me this is "the default correct behavior". I cannot argue against anything.

I only know that without availability the system breaks, and importing availability isn't in scope of this US

Reply:
This is actually the only possible behavior I can think of, without widening the scope of the US.

Reply:
I know it's more MVP of API but still, without availability, it's not usable to me.
And if I saw properly, handling this is not planed in any future US in the MVP Epic nor in the roadmap plans.

Adding an optional field for availability really widen the scope so much?

Reply:
Well... I would leave this discussion for BV, what do you think?

Reply:
or we can involve @LukasHeinz without waiting for BV!

Martin:
> I know it's more MVP of API but still, without availability, it's not usable to me.
And if I saw properly, handling this is not planed in any future US in the MVP Epic nor in the roadmap plans.

Petr:
> This is actually the only possible behavior I can think of, without widening the scope of the US.

Me...
* Handling availability is not part of this US. We cannot always solve everything
* If implementators needs to work with availability immediately, they can implement handling it by themselves
* Unfortunatelly it is more complicated how should be handled... string constrants? Is it enough? What about stocks? What if the implementators extend availabilities by new ones?
* So I vote for not solving availabilities in this US, and create a new US that will solve only them and do the US right after this one

Reply:
This PR is a first step of the import API, so anyone can use it. But there is more work to do like categories, prices and of course availability. So availability won't be addressed now

Reply:
Availability is literally only required field when you create a product. I appreciate if we push availabilities on the top of things to do in API :)